File: im-parent/mianshi-im-api/src/main/webapp/pages/console/js/payConfig.js
layui.use(['form','jquery',"layer"],function() {
    var form = layui.form,
        $ = layui.jquery,
        layer = parent.layer === undefined ? layui.layer : top.layer;

    //非管理员登录屏蔽操作按钮
    if(localStorage.getItem("IS_ADMIN")==0){
        $(".save").remove();
    }

    //获取当前系统支付配置
    Common.invoke({
        url : request('/console/payConfig'),
        data : {},
        successMsg : false,
        errorMsg : "获取数据失败,请检查网络",
        success : function(result) {
            fillParameter(result.data);
        },
        error : function(result) {
        }

    });

    form.on("submit(walletConfig)",function(){
        var payConfig = {};
        payConfig.id = 10000;
        payConfig.maxTransferAmount = $(".maxTransferAmount").val();
        payConfig.maxRedpacktAmount = $(".maxRedpacktAmount").val();
        payConfig.maxRedpacktNumber = $(".maxRedpacktNumber").val();
        payConfig.maxWithdrawAmount = $(".maxWithdrawAmount").val();
        payConfig.maxRechargeAmount = $(".maxRechargeAmount").val();
        payConfig.maxCodePaymentAmount = $(".maxCodePaymentAmount").val();
        payConfig.maxCodeReceiptAmount = $(".maxCodeReceiptAmount").val();
        payConfig.dayMaxTransferAmount = $(".dayMaxTransferAmount").val();
        payConfig.dayMaxRedpacktAmount = $(".dayMaxRedpacktAmount").val();
        payConfig.dayMaxWithdrawAmount = $(".dayMaxWithdrawAmount").val();
        payConfig.dayMaxRechargeAmount = $(".dayMaxRechargeAmount").val();
        payConfig.dayMaxCodePayAmount = $(".dayMaxCodePayAmount").val();
        payConfig.dayMaxCodeReceiptAmount = $(".dayMaxCodeReceiptAmount").val();
        payConfig.isOpenManualPay = $(".isOpenManualPay").val();
        payConfig.isOpenCloudWallet = $(".isOpenCloudWallet").val();
        payConfig.isDefaultFreeze = $(".isDefaultFreeze").val();
        payConfig.myChangeWithdrawRate = $(".myChangeWithdrawRate").val();
        payConfig.newYopUserRechargeTime = $(".newYopUserRechargeTime").val();
        payConfig.newYopUserFirstRecharge = $(".newYopUserFirstRecharge").val();
        payConfig.consoleMaxRechargeAmount = $(".consoleMaxRechargeAmount").val();
        payConfig.consoleMaxCodePaymentAmount = $(".consoleMaxCodePaymentAmount").val();
        payConfig.manualPaywithdrawFee = $(".manualPaywithdrawFee").val();
        payConfig.rechargePresentedScale = $(".rechargePresentedScale").val();
        payConfig.upayWithdrawServiceCharge = $(".upayWithdrawServiceCharge").val();
        payConfig.minWithdrawAmount = $(".minWithdrawAmount").val();
        payConfig.isOpenAuditPay = $(".isOpenAuditPay").val();
        payConfig.isOpenAutoPay = $(".isOpenAutoPay").val();
        payConfig.isOpenWXPay = $(".isOpenWXPay").val();
        payConfig.minRechargeAmount = $(".minRechargeAmount").val();
        payConfig.myChangeWithdrawBase = $(".myChangeWithdrawBase").val();
        payConfig.isOpenDepositWithdrawal = $(".isOpenDepositWithdrawal").val();
        payConfig.isOpenCZ = $(".isOpenCZ").val();
        if($(".manualPaywithdrawFee").val() == ""){
            layer.alert("提现费率不能为空");
            return;
        }else if($(".manualPaywithdrawFee").val() > 1){
            layer.alert("提现费率要小于1");
            return;
        }

        Common.invoke({
            url : request('/console/payConfig/set'),
            data : payConfig,
            successMsg : "操作成功",
            errorMsg : "操作失败,请检查网络",
            success : function(result) {

            },
            error : function(result) {

            }

        });
        return false;
    })

});

function fillParameter(data){
    //判断字段数据是否存在
    function nullData(data){
        if(data == '' || data == "undefined" || data==null){
            return "";
        }else{
            return data;
        }
    }
    // 数据回现
    $(".maxTransferAmount").val(data.maxTransferAmount);
    $(".maxRedpacktAmount").val(data.maxRedpacktAmount);
    $(".maxRedpacktNumber").val(data.maxRedpacktNumber);
    $(".maxWithdrawAmount").val(data.maxWithdrawAmount);
    $(".maxRechargeAmount").val(data.maxRechargeAmount);
    $(".maxCodePaymentAmount").val(data.maxCodePaymentAmount);
    $(".maxCodeReceiptAmount").val(data.maxCodeReceiptAmount);
    $(".dayMaxTransferAmount").val(data.dayMaxTransferAmount);
    $(".dayMaxRedpacktAmount").val(data.dayMaxRedpacktAmount);
    $(".dayMaxWithdrawAmount").val(data.dayMaxWithdrawAmount);
    $(".dayMaxRechargeAmount").val(data.dayMaxRechargeAmount);
    $(".dayMaxCodePayAmount").val(data.dayMaxCodePayAmount);
    $(".dayMaxCodeReceiptAmount").val(data.dayMaxCodeReceiptAmount);
    $(".isOpenManualPay").val(data.isOpenManualPay);
    $(".isOpenCloudWallet").val(data.isOpenCloudWallet);
    $(".isDefaultFreeze").val(data.isDefaultFreeze);
    $(".myChangeWithdrawRate").val(data.myChangeWithdrawRate);
    $(".newYopUserRechargeTime").val(data.newYopUserRechargeTime);
    $(".newYopUserFirstRecharge").val(data.newYopUserFirstRecharge);
    $(".consoleMaxRechargeAmount").val(data.consoleMaxRechargeAmount);
    $(".consoleMaxCodePaymentAmount").val(data.consoleMaxCodePaymentAmount);
    $(".manualPaywithdrawFee").val(data.manualPaywithdrawFee);
    $(".rechargePresentedScale").val(data.rechargePresentedScale);
    $(".upayWithdrawServiceCharge").val(data.upayWithdrawServiceCharge);
    $(".minWithdrawAmount").val(data.minWithdrawAmount);
    $(".isOpenAuditPay").val(data.isOpenAuditPay);
    $(".isOpenAutoPay").val(data.isOpenAutoPay);
    $(".isOpenWXPay").val(data.isOpenWXPay);
    $(".minRechargeAmount").val(data.minRechargeAmount);
    $(".myChangeWithdrawBase").val(data.myChangeWithdrawBase);
    $(".isOpenCZ").val(data.isOpenCZ);
    $(".isOpenDepositWithdrawal").val(data.isOpenDepositWithdrawal);
    //重新渲染
    layui.form.render();
}
$(function () {
    //权限判断
    var arr=['pay-sava'];
    manage.authButton(arr);
    //调用父级页面的Js函数
    window.parent.getJointVisitPath();
})

File: im-comm-modules/im-core-module/src/main/java/com/basic/im/entity/PayConfig.java
package com.basic.im.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * @author zhm
 * @version V1.0
 * @Description: TODO(todo)
 * @date 2020/1/4 9:43
 */
@Data
@Document(value="payConfig")
public class PayConfig {

    private @Id long id=10000;

    /**
     * 单笔最大转账金额
     */
    private double maxTransferAmount;

    /**
     * 单笔最大红包金额
     */
    private double maxRedpacktAmount=200.0;

    /**
     * 单笔群红包最大发送个数
     */
    private int maxRedpacktNumber = 10;

    /**
     * 单笔最大提现金额
     */
    private double maxWithdrawAmount;

    /**
     * 单笔最大充值金额
     */
    private double maxRechargeAmount;

    /**
     * 单笔付款码支付最大金额
     */
    private double maxCodePaymentAmount;

    /**
     * 单日付款码支付最大金额
     */
    private double dayMaxCodePayAmount;

    /**
     * 单笔二维码收款最大金额
     */
    private double maxCodeReceiptAmount;

    /**
     * 单日二维码收款最大金额
     */
    private double dayMaxCodeReceiptAmount;

    /**
     * 用户单日转账最大金额
     */
    private double dayMaxTransferAmount;

    /**
     * 用户单日红包最大金额
     */
    private double dayMaxRedpacktAmount;

    /**
     * 用户单日提现最大金额
     */
    private double dayMaxWithdrawAmount;

    /**
     * 用户单日充值最大金额
     */
    private double dayMaxRechargeAmount;

    /**
     * 零钱提现费率
     */
    private double myChangeWithdrawRate = 0.006;


    /**
     * 银行卡 提现手续费
     */
    private double upayWithdrawServiceCharge=0.1;

    /**
     * 是否开启扫码手动充值
     */
    private byte isOpenManualPay = 0;

    /**
     * 是否开启云钱包 1:开启 0:关闭
     */
    private byte isOpenCloudWallet = 0;

    /**
     * 云钱包开户用户是否默认不冻结
     */
    private byte isDefaultFreeze = 0;

    /**
     * 云钱包新开户充值多少小时后可交易
     */
    private double newYopUserRechargeTime = 24;

    /**
     * 云钱包新开户第一笔充值限额
     */
    private double newYopUserFirstRecharge = 10;

    /**
     * 后台管理中单笔最大充值金额
     */
    private double consoleMaxRechargeAmount = 1;

    /**
     * 后台管理中单笔最大手工提现金额
     */
    private double consoleMaxCodePaymentAmount = 1;

    /**
     *  手动充值功能-提现的费率
     **/
    private double manualPaywithdrawFee = 0.006;

    /**
     * 充值赠送比例
     **/
    private double rechargePresentedScale = 0;
    /**
     * 单笔最小提现金额
     */
    private double minWithdrawAmount=1000.0;
    /**
     * 提现基础费用
     */
    private double myChangeWithdrawBase = 1.0;
    /**
     * 是否开启审核提现
     */
    private byte isOpenAuditPay = 1;
    /**
     * 是否开启自动提现
     */
    private byte isOpenAutoPay = 1;
    /**
     * 是否开启微信提现
     */
    private byte isOpenWXPay = 0;
    /**
     * 单笔最小充值金额
     */
    private double minRechargeAmount = 0;

    private byte isOpenDepositWithdrawal = 1;

    private byte isOpenCZ = 1;
}
